File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
num = int(input())
if num%5 == 0:
    print(num//5)
else:
    count = 0
    while num>0:
        num-=3
        count+=1
        if num%5 == 0:
            count += num//5
            print(count)
            break
        elif num == 1 or num ==2:
            print(-1)
            break
        elif num == 0:
            print(count)
            break

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int num = scan.nextInt();
        int check = 0;

        while(num>=0){
            if(num%5 == 0){
                check += num/5;
                num %= 5;
                break;
            }
            num -= 3;
            check+=1;
        }
        if(num<0){
            System.out.println(-1);
        } else{
            System.out.println(check);

        }

    }
}
